src/common/HTTPRequest.ts

```diff
@@ -174,13 +176,12 @@ export class HTTPRequest {
     this._interceptionId = interceptionId;
     this._allowInterception = allowInterception;
     this._url = event.request.url;
-    this._resourceType = event.type.toLowerCase() as ResourceType;
+    this._resourceType = event.type?.toLowerCase() as ResourceType;
```

Comment:
Are we sure that `event.type?.toLowerCase()` will be `ResourceType`?

If so, can we add a comment to why we're sure?
If not, I think we need to update typing to reflect `_resourceType` being able to `undefined`. WDYT?

Reply:
I think in practice it's rarely empty if at all but on the CDP level is can be undefined.

Reply:
We cannot change it undefined because our API docs say it's always defined. I will add a fallback to the type `other`.